src/hooks/useNotifications.tsx

```diff
@@ -82,24 +98,34 @@ const notificationQuery = gql`
               }
             }
           }
+          ... on PrivatePostNotification {
+            post {
+              id
+              __typename
+              ... on ActivePost {
+                authorId
+              }
+            }
+          }
         }
       }
     }
   }
 `;
 
-const supportedNotificationTypes = [
-  'PostReplyNotification',
-  'CircleAdminPostNotification',
-  'SurveyAssignedNotification',
-];
+function isSupportedNotification(
+  __typename: string,
+): __typename is SupportedNotificationTypes {
+  return (__typename as SupportedNotificationTypes) !== undefined;
```

Comment:
I don't think this is doing what you want it to do. I think you will need either an enum, an array of strings, or an exhaustive switch to check if it is an expected type. It's probably okay for the moment though since only supported types are being queried.

Reply:
I'll write a unit test for this and update accordingly. I believed this to be correct based on [this documentation](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) but perhaps I'm misunderstanding how this works.

Reply:
You are correct, I misunderstood what that could do. Reverted just to use an array of supported type name strings.